service/src/main/java/uk/gov/hmcts/reform/fpl/service/RepresentativeService.java

```diff
@@ -0,0 +1,186 @@
+
+@Service
+public class RepresentativeService {
+
+    @Autowired
+    private OrganisationService organisationService;
+
+    @Autowired
+    private CaseService caseService;
+
+    @Autowired
+    private CaseDataExtractionService caseDataExtractionService;
+
+    public List<Element<Representative>> getDefaultRepresentatives(CaseData caseData) {
+        if (ObjectUtils.isEmpty(caseData.getRepresentatives())) {
+            return wrapElements(Representative.builder().build());
+        } else {
+            return caseData.getRepresentatives();
+        }
+    }
+
+    public List<String> validateRepresentatives(CaseData caseData, String authorisation) {
+        List<String> validationErrors = new ArrayList<>();
+
+        List<Representative> representatives = unwrapElements(caseData.getRepresentatives());
+
+        int representativeSequence = 1;
+
+        for (Representative representative : representatives) {
+
+            final RepresentativeServingPreferences servingPreferences = representative.getServingPreferences();
+            final RepresentativeRole role = representative.getRole();
+
+            String representativeLabel = representatives.size() == 1
+                ? "Representative" : "Representative " + representativeSequence++;
+
+            if (isEmpty(representative.getFullName())) {
+                validationErrors.add(format("Enter a full name for %s", representativeLabel));
+            }
+
+            if (isEmpty(representative.getPositionInACase())) {
+                validationErrors.add(format("Enter a position in the case for %s", representativeLabel));
+            }
+
+            if (isNull(role)) {
+                validationErrors.add(format("Select who %s is", representativeLabel));
+            }
+
+            if (isNull(servingPreferences)) {
+                validationErrors.add(format("Select how %s wants to get case information", representativeLabel));
+            }
+
+            if (EMAIL.equals(servingPreferences)) {
+                if (isEmpty(representative.getEmail())) {
+                    validationErrors.add(format("Enter an email address for %s", representativeLabel));
+                }
+            }
+
+            if (POST.equals(servingPreferences)) {
+                if (isNull(representative.getAddress()) || isEmpty(representative.getAddress().getPostcode())) {
+                    validationErrors.add(format("Enter a postcode for %s", representativeLabel));
+                }
+                if (isNull(representative.getAddress()) || isEmpty(representative.getAddress().getAddressLine1())) {
+                    validationErrors.add(format("Enter a valid address for %s", representativeLabel));
+                }
+            }
+
+            if (nonNull(role) && RESPONDENT.equals(representative.getRole().getType())) {
+                Optional<Representable> responded = findRespondent(caseData, role.getSequenceNo());
```

Comment:
looks like a typo here, `respondent`, but this variable can be removed

Reply:
done